### ISSUE TITLE
Change Request: `Linter.getRules()` equivalent support in Flat ESLint

### ISSUE THREAD

**Issue by snitin315** [eslint/eslint#18103]
### ESLint version

v9 prerelease

### What problem do you want to solve?

In FlatESLint, `Linter.getRules()` is not available. 

In the playground implementation, we were using [this method to display available rules](https://github.com/eslint/eslint.org/blob/56e2580e4ea9da16ba28bb6d7a8ee169bcd3ca15/src/playground/App.js#L17) in the sidebar.

As this method is no longer available in Flat Config, we had to create a new instance of LegacyEslint just to use `Linter.getRules()`

```js
const linter = new Linter({ configType: "flat" });
const legacyLinter = new Linter({ configType: "eslintrc" });
const rules = legacyLinter.getRules();
```

### What do you think is the correct solution?

We need an equivalent API for`Linter.getRules()` in Flat ESLint.

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by fasttime** [comment]
Do you need the complete implementation for each rule, or would the rule names and `meta` data be sufficient?

**Comment by snitin315** [comment]
Only rule name & metadata should be sufficient.

**Comment by fasttime** [comment]
All information is included in a JSON file in the docs package: https://github.com/eslint/eslint/blob/main/docs/src/_data/rules_meta.json, but I'm not sure if that file can be practically used to build the Playground.

**Comment by nzakas** [comment]
Yes, removing `getRules()` was intended because the rules it returned varied based on when the method was called. If it was after a file was linted with plugin rules, then you'd get core rules plus plugin rules, otherwise you'd just get core rules. Fundamentally, there's no need to access the raw rule objects, so we didn't include a replacement.

Because we only need the rules meta, I think @fasttime's suggestion of using the data file from the docs site makes sense. It seems like maybe we just need the rule ID? Or is there something more we need?

**Comment by snitin315** [comment]
Yeah, we can use the docs data file, just need `ruleId` & `rule.deprecated` properties. We can use the `latest` branch to make sure we always get published data.

**Comment by nzakas** [comment]
Sounds good@

**Comment by aladdin-add** [comment]
seems we have reached consensus to just use the docs data file. so the issue can be closed? @snitin315

**Comment by snitin315** [comment]
Yes, we can close this.
